app/models/host/managed.rb

```
@@ -796,10 +796,13 @@ def self.report_status
   # if we are managing DNS, we can query the correct DNS server
   # otherwise, use normal systems dns settings to resolv
   def to_ip_address name_or_ip
+    return '127.0.0.1' if name_or_ip == 'localhost'
     return name_or_ip if name_or_ip =~ Net::Validations::IP_REGEXP
     return dns_ptr_record.dns_lookup(name_or_ip).ip if dns_ptr_record
     # fall back to normal dns resolution
     domain.resolver.getaddress(name_or_ip).to_s
+  rescue
+    raise ::Foreman.Exception.new(N_("Unable to detect TFTP server IP address - override in proxy config"))
```

I would also log the exception and not swallow it compleltly